internal/resources/k6/resource_schedule.go

```diff
@@ -0,0 +1,545 @@
+	Until     types.String   `tfsdk:"until"`
+	Byday     []types.String `tfsdk:"byday"`
+}
+
+// scheduleResourceModel maps the resource schema data.
+type scheduleResourceModel struct {
+	ID             types.String         `tfsdk:"id"`
+	LoadTestID     types.String         `tfsdk:"load_test_id"`
+	Starts         types.String         `tfsdk:"starts"`
+	RecurrenceRule *recurrenceRuleModel `tfsdk:"recurrence_rule"`
+	Deactivated    types.Bool           `tfsdk:"deactivated"`
+	NextRun        types.String         `tfsdk:"next_run"`
+	CreatedBy      types.String         `tfsdk:"created_by"`
+}
+
+// scheduleResource is the resource implementation.
+type scheduleResource struct {
+	basePluginFrameworkResource
+}
+
+// Metadata returns the resource type name.
+func (r *scheduleResource) Metadata(_ context.Context, _ resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = resourceScheduleName
+}
+
+// Schema defines the schema for the resource.
+func (r *scheduleResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages a k6 schedule for automated test execution.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Numeric identifier of the schedule.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"load_test_id": schema.StringAttribute{
+				Description: "The identifier of the load test to schedule.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"starts": schema.StringAttribute{
+				Description: "The start time for the schedule (RFC3339 format).",
+				Required:    true,
+			},
+			"deactivated": schema.BoolAttribute{
+				Description: "Whether the schedule is deactivated.",
+				Computed:    true,
+			},
+			"next_run": schema.StringAttribute{
+				Description: "The next scheduled execution time.",
+				Computed:    true,
+			},
+			"created_by": schema.StringAttribute{
+				Description: "The email of the user who created the schedule.",
+				Computed:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"recurrence_rule": schema.SingleNestedBlock{
+				Description: "The schedule recurrence settings. If not specified, the test will run only once on the 'starts' date.",
+				Attributes: map[string]schema.Attribute{
+					"frequency": schema.StringAttribute{
+						Description: "The frequency of the schedule (HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY).",
+						Required:    true,
+						Validators: []validator.String{
+							stringvalidator.OneOf("HOURLY", "DAILY", "WEEKLY", "MONTHLY", "YEARLY"),
+						},
+					},
+					"interval": schema.Int32Attribute{
+						Description: "The interval between each frequency iteration (e.g., 2 = every 2 hours for HOURLY). Defaults to 1.",
+						Optional:    true,
+					},
```

Comment:
This value defaults to `null` in the provider and to `1` in the API, which leads to the following error:
```
Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to grafana_k6_schedule.k6_tf_demo_k6_schedule, provider "provider[\"registry.terraform.io/grafana/grafana\"].k6" produced an unexpected new value:
│ .recurrence_rule.interval: was null, but now cty.NumberIntVal(1).
```

To fix this, we should set the correct default in the provider.
```suggestion
						Optional:    true,
						Computed:    true,
						Default:      int32default.StaticInt32(1),
					},
```